mongodump/oplog_dump.go

```diff
@@ -125,16 +126,37 @@ func (dump *MongoDump) checkOplogTimestampExists(ts primitive.Timestamp) (bool,
 
 func oplogDocumentValidator(in []byte) error {
 	raw := bson.Raw(in)
-	if nsVal, err := raw.LookupErr("ns"); err == nil {
-		if nsStr, ok := nsVal.StringValueOK(); ok && nsStr == "admin.system.version" {
-			return fmt.Errorf("cannot dump with oplog if admin.system.version is modified")
-		}
+
+	var nsStr string
+	var ok bool
+
+	nsVal, err := raw.LookupErr("ns")
+
+	if err == nil {
+		nsStr, ok = nsVal.StringValueOK()
+	}
+
+	if ok && nsStr == "admin.system.version" {
+		return fmt.Errorf("cannot dump with oplog if admin.system.version is modified")
 	}
 
 	if _, err := raw.LookupErr("o", "renameCollection"); err == nil {
 		return fmt.Errorf("cannot dump with oplog while renames occur")
 	}
 
+	if ok {
+		dbName := strings.SplitN(nsStr, ".", 2)[0]
+
+		if dbName == "config" {
+			if collNameRaw, err := raw.LookupErr("o", "create"); err == nil {
```

Comment:
should we also be checking `insert` commands?

Reply:
Oh yeah, I put that in the design to cover all the bases. But I think just checking for existence at the start and create oplog entries is sufficient, right? I don't see how we would miss anything with the checks we currently have.

Reply:
Resolving per discussion in standup this morning about how this should be sufficient, because collections created by an insert are still replicated as create+insert